llvm/lib/Target/Mips/Mips64InstrInfo.td

```diff
@@ -830,6 +830,12 @@ def : MipsPat<(sra GPR64:$rt, (i32 (trunc GPR64:$rs))),
 def : MipsPat<(rotr GPR64:$rt, (i32 (trunc GPR64:$rs))),
               (DROTRV GPR64:$rt, (EXTRACT_SUBREG GPR64:$rs, sub_32))>,
               ISA_MIPS3, GPR_64;
+def : MipsPat<(and (srl (i32 (trunc GPR64:$src)), immZExt5:$imm5), immZExt5:$value),
```

Comment:
Can the `and` immediate be larger than 5 bits? I'm not very familiar with Mips encodings.

Reply:
> Can the `and` immediate be larger than 5 bits? I'm not very familiar with Mips encodings.

Sorry, it can be larger than 5, it should be `immZExt16`. I would submit again. Thanks!